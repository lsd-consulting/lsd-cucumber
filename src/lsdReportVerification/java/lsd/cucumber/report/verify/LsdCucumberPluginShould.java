package lsd.cucumber.report.verify;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static java.nio.file.Paths.get;
import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/lsd-index.html"));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario.html"));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Data_table.html"));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario_outline.html"));
    }

    private File copyOfFile(String path) throws IOException {
        Path original = get(path);
        Path copy = Path.of(original.getParent().toString(), "copy_" + original.getFileName());
        Files.copy(original, copy, REPLACE_EXISTING);
        return copy.toFile();
    }
}
