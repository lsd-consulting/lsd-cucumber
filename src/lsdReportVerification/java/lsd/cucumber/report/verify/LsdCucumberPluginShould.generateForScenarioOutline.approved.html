<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>scenario_outline</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
<main role="main">
    <h1 class="logo">LSD</h1>
    <h1>scenario_outline</h1>
    <div class="dropdown">
        <label class="dropdown">Scenarios</label>
        <div class="dropdown-content">
            <a class="success" href="#7">Scenario with a data table #1</a>
            <a class="success" href="#10">Scenario with a data table #2</a>
        </div>
    </div><article id="7">
        <span onclick="location.href='#top';" title="top of page" class="back_to_top"></span>
        <h2 class="success">Scenario with a data table #1</h2><section class="description success">
            <h3 class="success">Description</h3>
            Given  the following values for value1_1 and value1_2</br>When  when3</br>Then  then3
        </section>
        
        <section class="sequence diagram svg success">
            <h3 class="success">Sequence Diagram</h3>
            <figure><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="152px" preserveAspectRatio="none" style="width:212px;height:152px;" version="1.1" viewBox="0 0 212 152" width="212px" zoomAndPan="magnify"><defs/><g><line style="stroke:#000000;stroke-width:1.0;" x1="51" x2="51" y1="37.0146" y2="116.6133"/><line style="stroke:#000000;stroke-width:1.0;" x1="159" x2="159" y1="37.0146" y2="116.6133"/><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="5"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="26.0752">Source</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="115.6133"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="136.6885">Source</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="5"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="26.0752">Destination</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="115.6133"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="136.6885">Destination</text><polygon fill="#000000" points="147,64.814,157,68.814,147,72.814,151,68.814" style="stroke:#000000;stroke-width:1.0;"/><line style="stroke:#000000;stroke-width:1.0;" x1="51" x2="153" y1="68.814" y2="68.814"/><a href="#id1" target="_top" title="label1" xlink:actuate="onRequest" xlink:href="#id1" xlink:show="new" xlink:title="label1" xlink:type="simple"><text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="58" y="64.0845">label1</text></a><polygon fill="#000000" points="62,94.6133,52,98.6133,62,102.6133,58,98.6133" style="stroke:#000000;stroke-width:1.0;"/><line style="stroke:#000000;stroke-width:1.0;" x1="56" x2="158" y1="98.6133" y2="98.6133"/><a href="#id2" target="_top" title="label2" xlink:actuate="onRequest" xlink:href="#id2" xlink:show="new" xlink:title="label2" xlink:type="simple"><text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="68" y="93.8838">label2</text></a></g></svg></figure>
            <div onclick="location.href='#!';" id="8" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code8')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code8">@startuml

!theme plain

skinparam backgroundColor transparent

hide unlinked






Source -&gt; Destination: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id1 {label1} label1]]&lt;/text&gt;

Destination -&gt; Source: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id2 {label2} label2]]&lt;/text&gt;


@enduml</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#8">[source]</a></sub>
            <sub><a href="#9">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="9" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="153px" preserveAspectRatio="none" style="width:137px;height:153px;" version="1.1" viewBox="0 0 137 153" width="137px" zoomAndPan="magnify"><defs/><g><!--entity Source--><g id="elem_Source"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="88" x="27" y="17"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="95" y="22"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="24"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="28"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="42" y="51.0752">Source</text></g><!--entity Destination--><g id="elem_Destination"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="120" x="11" y="100"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="111" y="105"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="107"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="111"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="26" y="134.0752">Destination</text></g><path d="M64.8766,64.063 C64.0451,75.204 64.0411,88.6144 64.8648,99.7771 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="64.8648,99.7771,68.1917,90.5071,64.4968,94.7906,60.2134,91.0958,64.8648,99.7771" style="stroke:#000000;stroke-width:1.0;"/><path d="M77.1352,99.7771 C77.9589,88.6144 77.9549,75.204 77.1234,64.063 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="77.1234,64.063,73.8043,73.3358,77.4955,69.0491,81.7821,72.7403,77.1234,64.063" style="stroke:#000000;stroke-width:1.0;"/></g></svg></pre>
                </div>
                <sub><a href="#uml9">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml9" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code9')"></span>
                <a class="close" href="#9">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code9">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent

component Source
component Destination


[Source] --&gt; Destination
[Destination] --&gt; Source


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups --><div onclick="location.href='#!';" id="id1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label1</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div><div onclick="location.href='#!';" id="id2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label2</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
    </article><article id="10">
        <span onclick="location.href='#top';" title="top of page" class="back_to_top"></span>
        <h2 class="success">Scenario with a data table #2</h2><section class="description success">
            <h3 class="success">Description</h3>
            Given  the following values for value2_1 and value2_2</br>When  when3</br>Then  then3
        </section>
        
        <section class="sequence diagram svg success">
            <h3 class="success">Sequence Diagram</h3>
            <figure><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="152px" preserveAspectRatio="none" style="width:212px;height:152px;" version="1.1" viewBox="0 0 212 152" width="212px" zoomAndPan="magnify"><defs/><g><line style="stroke:#000000;stroke-width:1.0;" x1="51" x2="51" y1="37.0146" y2="116.6133"/><line style="stroke:#000000;stroke-width:1.0;" x1="159" x2="159" y1="37.0146" y2="116.6133"/><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="5"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="26.0752">Source</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="115.6133"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="136.6885">Source</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="5"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="26.0752">Destination</text><rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="115.6133"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="136.6885">Destination</text><polygon fill="#000000" points="147,64.814,157,68.814,147,72.814,151,68.814" style="stroke:#000000;stroke-width:1.0;"/><line style="stroke:#000000;stroke-width:1.0;" x1="51" x2="153" y1="68.814" y2="68.814"/><a href="#id1" target="_top" title="label1" xlink:actuate="onRequest" xlink:href="#id1" xlink:show="new" xlink:title="label1" xlink:type="simple"><text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="58" y="64.0845">label1</text></a><polygon fill="#000000" points="62,94.6133,52,98.6133,62,102.6133,58,98.6133" style="stroke:#000000;stroke-width:1.0;"/><line style="stroke:#000000;stroke-width:1.0;" x1="56" x2="158" y1="98.6133" y2="98.6133"/><a href="#id2" target="_top" title="label2" xlink:actuate="onRequest" xlink:href="#id2" xlink:show="new" xlink:title="label2" xlink:type="simple"><text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="68" y="93.8838">label2</text></a></g></svg></figure>
            <div onclick="location.href='#!';" id="11" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code11')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code11">@startuml

!theme plain

skinparam backgroundColor transparent

hide unlinked






Source -&gt; Destination: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id1 {label1} label1]]&lt;/text&gt;

Destination -&gt; Source: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id2 {label2} label2]]&lt;/text&gt;


@enduml</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#11">[source]</a></sub>
            <sub><a href="#12">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="12" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="153px" preserveAspectRatio="none" style="width:137px;height:153px;" version="1.1" viewBox="0 0 137 153" width="137px" zoomAndPan="magnify"><defs/><g><!--entity Source--><g id="elem_Source"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="88" x="27" y="17"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="95" y="22"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="24"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="28"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="42" y="51.0752">Source</text></g><!--entity Destination--><g id="elem_Destination"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="120" x="11" y="100"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="111" y="105"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="107"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="111"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="26" y="134.0752">Destination</text></g><path d="M64.8766,64.063 C64.0451,75.204 64.0411,88.6144 64.8648,99.7771 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="64.8648,99.7771,68.1917,90.5071,64.4968,94.7906,60.2134,91.0958,64.8648,99.7771" style="stroke:#000000;stroke-width:1.0;"/><path d="M77.1352,99.7771 C77.9589,88.6144 77.9549,75.204 77.1234,64.063 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="77.1234,64.063,73.8043,73.3358,77.4955,69.0491,81.7821,72.7403,77.1234,64.063" style="stroke:#000000;stroke-width:1.0;"/></g></svg></pre>
                </div>
                <sub><a href="#uml12">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml12" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code12')"></span>
                <a class="close" href="#12">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code12">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent

component Source
component Destination


[Source] --&gt; Destination
[Destination] --&gt; Source


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups --><div onclick="location.href='#!';" id="id1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label1</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div><div onclick="location.href='#!';" id="id2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label2</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
    </article>
</main>
</body>
<footer>
    <script type="text/javascript" src="https://d3js.org/d3.v4.min.js"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/styles/default.min.css">
    <script src="custom.js"></script>
    <script>
        addSliderForSvgZoom();
        highlightLifelinesWhenClicked();
    </script>
    <script src="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/highlight.min.js"></script>
    <script>hljs.highlightAll();</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"></script>
    <script>
        highlightKeywords();
        unMarkSvg();
    </script>
</footer>
</html>
